Plugins/Race/Race.cpp

```diff
@@ -178,6 +178,24 @@ void Race::DoEffect(CNWSCreature *pCreature,
     pCreature->ApplyEffect(eff, true, true);
 }
 
+void Race::RemoveRaceEffects(CNWSCreature *pCreature)
+{
+    auto effectsLevelAdded = *pCreature->nwnxGet<int>("RACEMODS_ADDED_LEVEL");
+    if (pCreature->m_pStats == nullptr || effectsLevelAdded == 0)
+        return;
+
+    for (int i = pCreature->m_appliedEffects.num; i >= 0; --i)
+    {
+        auto eff = (CGameEffect*)pCreature->m_appliedEffects.element[i];
+        if (eff->m_sCustomTag == "NWNX_Race_RacialMod")
+        {
+            pCreature->RemoveEffectById(eff->m_nID);
+        }
+    }
```

Comment:
Deleting an effect can remove multiple linked ones so this would have a big chance for out of bound array access, you can't trust the effect list after a removal. The previous method of adding IDs to be removed to a list is better

Reply:
Ah of course. I had reasoned that, under the old method, RemoveEffectById would try and remove invalid effects anyway, so it /must/ be fine to attempt to do so. Didn't think through the array access. New commit reverts this section to the previous method.